### DISCUSSION TITLE
启动顺序问题

### DISCUSSION THREAD

**Discussion by snakwu** [nextcloud/all-in-one#4097] [Questions] [answered]
当用docker compose -d 正常安装nextcloud-aio后，一切正常后，如果重启宿主机，nextcloud服务会自动启动，当nextcloud服务启动后，nextcloud-aio服务页面就无法正常启动，必须要停止所有nextcloud的相关服务，再重新启动nextcloud-aio，再在nextcloud-aio的服务页面去启动所有容器才正常。简单讲，就是当如果nextcloud的容器启动后，nextcloud-aio就无法启动.

能否加个判断，如果nextcloud-aio-mastercontainer没有启动，nextcloud-aio-nextcloud等服务就停止不启动

**Comment by szaimen** [comment]
Hi, please report any issue in english. Otherwise I cannot help you.

**Comment by snakwu** [comment]
@szaimen sorry My English is poor

**Comment by snakwu** [comment]
When nextcloud-aio is installed normally with docker compose-d, if the host is restarted, the nextcloud service will start automatically. When the nextcloud service is started, the nextcloud-aio service page will not start normally. You must stop all nextcloud-related services, restart nextcloud-aio, and then start all containers on the nextcloud-aio service page. To put it simply, if nextcloud's container starts, nextcloud-aio will not start.

Can you add a judgment: if nextcloud-aio-mastercontainer is not started, services such as nextcloud-aio-nextcloud will stop and not start?

**Comment by szaimen** [comment]
> Can you add a judgment: if nextcloud-aio-mastercontainer is not started, services such as nextcloud-aio-nextcloud will stop and not start?

If the containers are not running while the server reboots, they will also not be started until you open the AIO interface and start them from there. However if they are running, they will also be started automatically when the server reboots.

**Comment by snakwu** [comment]
@szaimen I mean, when the system is rebooted, nextcloud-aio-nextcloud can start automatically, but nextcloud-aio-mastercontainer can't.

**Reply by szaimen** [reply]
The mastercontainer should always start after a reboot due to this:
![image](https://github.com/nextcloud/all-in-one/assets/42591237/70bcec24-a788-469b-8250-7949d2826e6b)

**Reply by snakwu** [reply]
No, because when the system is restarted, services such as nextcloud-aio-nextcloud are started first, and then nextcloud-aio-mastercontainer is started, there will be conflicts because nextcloud-aio-net has been established, so nextcloud-aio-mastercontainer startup fails

**Reply by szaimen** [reply]
What exactly is conflicting? Can you maybe post some logs?

**Reply by snakwu** [reply]
It is a network conflict, because after rebooting the system, nextcloud-aio-nextcloud will establish a network called nextcloud-aio when it starts, and then when nextcloud-aio-mastercontainer starts, it will also build this network, but this network already exists, so if there is a conflict, it cannot be started.

**Reply by szaimen** [reply]
Hm... Usually it should only try to rebuild this nrtwork if you open the aio interface. Also any issue of duplicate networks should be handled correctly and not run into this issue. Do you maybe have some logs availabke that show the problem?

**Reply by snakwu** [reply]
Jan 23 06:57:04 nextcloud-server dockerd[915]: time="2024-01-23T06:57:04.217064490+08:00" level=error msg="failed to start container"
container=303edbbb7ed4ce4b56d2ceec45322bdcea22f7eba53c7c48e77f77492b9b4d34 error="Address already in use"



Nextcloud-aio is already installed, and then restart the system, this will happen. At present, nextcloud-aio-nextcloud is normal to start itself, nextcloud-aio-mastercontainer can not start, report the above error

**Reply by szaimen** [reply]
>  error="Address already in use"

Hm... which address is already in use?

BTW, I just tried restarting my test server and both nextcloud-aio-nextcloud and the mastercontainer come up without any problem...

**Reply by szaimen** [reply, marked as answer]
I guess you might be running into this: https://github.com/docker/for-linux/issues/1481

**Reply by snakwu** [reply]
It's possible, but when I stop all nextcloyd-aio-xxx services and docker compose up-d, everything will be fine again, and it will appear again the next time the system is rebooted.
